src/sesclient.native/README.md

```diff
@@ -1,18 +1,65 @@
+| --application | Mandatory | Unique identifier for the application being requested.                                                                                                                                                                |
+| --thumbprint  | Optional  | Thumbprint of an additional certificate to accept in the TLS certificate chain of the HTTPS connection. <br/> **Note**: cannot be the thumbprint of a root certificate. <br/> Mandatory if `--common-name` specified. |
+| --common-name | Optional  | The common name of the certificate indicated by `--thumbprint`. <br/> Mandatory if `--thumbprint` specified.                                                                                                          |
+
+## Examples
+
+### Production
+
+In the Azure Batch production environment, `sesclient.native` can be used as a stand-in for any other application that has been integrated with our software entitlements service. This gives you or the Azure Batch team a diagnostic tool that can be used to isolate issues.
+
+To run within an Azure Batch Task:
+
+``` cmd
+sesclient.native --url %AZ_BATCH_ACCOUNT_URL% --token %AZ_BATCH_SOFTWARE_ENTITLEMENT_TOKEN% --application contosoapp
+```
+
+Substitute the name of your own application (or the one you are testing with) for `contosoapp`.
+
+Note that the environment variables `AZ_BATCH_ACCOUNT_URL` and `AZ_BATCH_SOFTWARE_ENTITLEMENT_TOKEN` are published by Azure Batch for use by a specific Task; they are **not** part of the environment available if/when you remote into a compute node.
+
+In this scenario, `sesclient.native` will verify that the connection is made to a genuine Azure Batch server by checking the certificate used to secure the HTTPS connection. If the certificate used for the connection does not trace back to one of the well known Microsoft intermediate certificate authorities built into the [native-code library](../Microsoft.Azure.Batch.SoftwareEntitlement.Client.Native), the software entitlement check will fail.
+
+### Testing
+
+In a local (non-Batch) test environment, you won't have access to a certificate signed by any one of the well known Microsoft intermediate certificate authorities, so `sesclient.native` allows you to specify your own certificate to use. 
+
+You'll need to specify the same certificate for `sesclient.native` as you already do for `sestest server` so the two components can handshake the connection properly.
+
+Assuming the appropriate thumbprint is available in the environment variable `%CONNECTION_THUMBPRINT%`, and the common name of that certificate is available in `%COMMON_NAME%`, run `sestest` as a server in a different console window:
+
+``` cmd
+sestest server --connection %CONNECTION_THUMBPRINT% --common-name ...
```

Comment:
You might as well mention %COMMON_NAME% since you go to the trouble of defining it *grin*